packages/vscode-extension/lib/wrapper.js

```diff
@@ -47,35 +51,34 @@ export function PreviewAppWrapper({ children, ...rest }) {
   const handleNavigationChange = useCallback(
     (navigationDescriptor) => {
       navigationHistory.set(navigationDescriptor.id, navigationDescriptor);
-      devtoolsAgent &&
-        devtoolsAgent._bridge.send("RNIDE_navigationChanged", {
-          displayName: navigationDescriptor.name,
-          id: navigationDescriptor.id,
-        });
+      devtoolsAgent?._bridge.send("RNIDE_navigationChanged", {
+        displayName: navigationDescriptor.name,
+        id: navigationDescriptor.id,
+      });
     },
     [devtoolsAgent]
   );
 
-  const useNavigationMainHook =
-    (navigationPlugins.length && navigationPlugins[0].plugin.mainHook) || emptyNavigationHook;
-  const { requestNavigationChange } = useNavigationMainHook({
+  const useNavigationMainHook = navigationPlugins[0]?.plugin.mainHook || emptyNavigationHook;
+  const { requestNavigationChange, onRouterInitialization } = useNavigationMainHook({
     onNavigationChange: handleNavigationChange,
   });
 
   const openPreview = useCallback(
     (previewKey) => {
-      AppRegistry.runApplication(previewKey, {
+      currentPreviewKey = previewKey;
+      AppRegistry.runApplication(PREVIEW_APP_KEY, {
         rootTag,
-        initialProps: {},
+        initialProps: { previewKey },
       });
+      const preview = global.__RNIDE_previews.get(previewKey);
+      handleNavigationChange({ id: previewKey, name: `preview:${preview.name}` });
     },
-    [rootTag]
+    [rootTag, devtoolsAgent]
   );
 
   const closePreview = useCallback(() => {
-    const SceneTracker = require("react-native/Libraries/Utilities/SceneTracker");
-    const isRunningPreview = isPreviewUrl(SceneTracker.getActiveScene().name);
-    if (isRunningPreview) {
+    if (currentPreviewKey) {
```

Comment:
maybe check active scene name, we don't need the exact preview name here as we only check if its running or not